pkg/util/version/version.go

```diff
@@ -14,12 +14,12 @@ type Version struct {
 }
 
 func NewVersion(s string) (version *Version, err error) {
+	// TODO[md]: handle prerelease versions (0.1.2-rc1) so they aren't appended to the previous component
+	// todo[md]: tbh just switch to hashicorp/go-version or github.com/Masterminds/semver/v3
```

Comment:
💯 ive used semver/v3 in the past, its pretty nice

Reply:
I always go to that one too, but I found it the other day it doesn't support "invalid" semver input like ubuntu's "22.04" with leading zeros. I was hoping to use the same version code in cog and the new base image generator code 😒